src/dataframe/dataframe.jl

```diff
@@ -640,9 +641,48 @@ function insertcols!(df::DataFrame, col_ind::Int, name_cols::Pair{Symbol,<:Any}.
                             "$(ncol(df)) columns at index $col_ind"))
     end
 
+    if !makeunique
+        if !allunique(first.(name_cols))
+            throw(ArgumentError("Names of columns to be inserted into a data frame must be " *
+                                "unique when `makeunique=true`"))
+        end
+        for (n, _) in name_cols
+            if hasproperty(df, n)
+                throw(ArgumentError("Column $n is already present in the data frame " *
+                                "which is not allowed when `makeunique=true`"))
+            end
+        end
+    end
+
+    if ncol(df) == 0
+        target_row_count = -1
+    else
+        target_row_count = nrow(df)
+    end
+
+    for (n, v) in name_cols
+        if v isa AbstractVector
+            if target_row_count == -1
+                target_row_count = length(v)
+            elseif length(v) != target_row_count
+                throw(DimensionMismatch("length of new column $n which is $(length(v)) must match" *
+                                        " the target number of rows in data frame ($target_row_count)"))
```

Comment:
This error may be a bit weird if you try to add two columns with different lengths to a zero-column data frame. Maybe check above the loop that all columns have the same size? Then you can just say here "number of rows" without the notion of target.

Reply:
OK - I will change it (but at least you seem to have gotten the meaning of `"target"` correctly here 😄)